File: java/morse.java
import java.util.*;
import java.lang.Object;
public class morse {
    static String space_between_words = "       ";// 7 blank units
    static String space_between_letters = "   "; // 3 blank units
    public static String remove_space_in_front(String n) {
        return n.substring(1);
    }

    public static String morse_to_normal(String n, Dictionary database) {
        // working....
        String n1=remove_space_in_front(n);
        String final_ans="";
        String[] split_into_words=n1.split(space_between_words);
        for (int t=0;t<split_into_words.length;t++){
            String[] split_into_letters=split_into_words[t].split(space_between_letters);
        for(String tmp_letter:split_into_letters){
            try{
                final_ans+=database.get(tmp_letter);
            }catch(Exception e){
                // do nothing...
            }
    }
    final_ans+=" ";
    }
    try{
        final_ans=final_ans.replace("null"," ");
    }catch(Exception e){
        // do nothing..
    }
        return final_ans;
    }

    public static String normal_to_morse(String n, Dictionary database) {
        // working...
        String final_ans="";
        String n1=remove_space_in_front(n);
        int len_of_string=n1.length();
        for(int itr=0;itr<len_of_string;itr++){
            String tmp_letter=""+n1.charAt(itr);
            final_ans+=(tmp_letter==" ")?"    ":(database.get(tmp_letter)+space_between_letters);
        }
        try{
            final_ans=final_ans.replace("null","");
        }
        catch(Exception e){
            // do nothing...
        }
        return final_ans;
    }

    public static void main(String d[]) {
        // driver code

         String characters[] = { "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q",
        "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "1", "2", "3", "4", "5", "6", "7", "8", "9", "0" };
 String morse[] = { ". _", "_ . . .", "_ . _ .", "_ . .", ".", ". . _ .", "_ _ .", ". . . .", ". .",
        ". _ _ _", "_ . _", ". _ _ _", "_ _", "_ .", "_ _ _", ". _ _ .", "_ _ . _", ". _ .", ". . .", "_", ". . _",
        ". . . _", ". __", "_ . . _", "_ . _ _", "_ _ . .", ". _ _ _ _", ". ._ _ _", ". . . _ _", ". . . . _",
        ". . . . .", "_ . . . .", "_ _ . . .", "_ _ _ . .", "_ _ _ _ .", "_ _ _ _ _" };



        Scanner nrt = new Scanner(System.in);
        Dictionary<String ,String> database1=new Hashtable<String ,String>();
        Dictionary<String ,String> database2=new Hashtable<String ,String>();

        // morse(key) ->  characters(value)
        for (int itr=0;itr<characters.length;itr++){
            database1.put(morse[itr],characters[itr]);
        }

        // characters(key)   -> morse(value)
        for(int itr=0;itr<morse.length;itr++){
            database2.put(characters[itr],morse[itr]);
        }
        while (true) {
            System.out.println(
                    "1: morse to normal\n2:normal to morse\n3: exit\n   enter information like this\n  1 <space > message");
            int user_choice = nrt.nextInt();
            String tmp;
            switch (user_choice) {
                case 1:
                    tmp = nrt.nextLine();
                    System.out.println(morse_to_normal(tmp,database1));
                    break;
                case 2:
                    tmp = nrt.nextLine();
                    System.out.println(normal_to_morse(tmp.toUpperCase(),database2));
                    break;
            }
            if (user_choice == 3) {
                break;
            }
        }
        nrt.close();
    }
}

File: python/morse.py



def morse_to_normal(user_str,database,s_b_w,s_b_l):
    # working....
    final_ans=""
    split_string_into_words=user_str.split(s_b_w)
    for tmp_word in split_string_into_words:
        split_string_into_letters=tmp_word.split(s_b_l)
        for letters in split_string_into_letters:
            final_ans+=str(database[letters])
        final_ans+=str(" ")
    

    # test code
    # for tmp_word in split_string_into_words:
    #     print("word>>",tmp_word)
    #     # words are separated fine...
    #     split_letters=tmp_word.split("   ")
    #     for tmp_letter in split_letters:
    #         print("letter> ",tmp_letter)
    return final_ans
def normal_to_morse(user_str,database):
    # working..
    final_ans=""
    for letter in user_str:
        try:
            if letter==str(" "):
                final_ans+=str("    ")
            else:
                final_ans+=str(database[letter])+str("   ")
        except KeyError:
            print(final_ans)
    return final_ans
# driver code
print("1: morse to normal\n2: normal to morse\n3: exit\n in the next line eneter your message")
user_choice=int(input())
user_msg=input()
space_between_words="       " #7 units
space_between_letters="   "#3 units
dataBase2={
    # keyname : value
    "A":". _",
    "B":"_ . . .",
    "C":"_ . _ .",
    "D":"_ . .",
    "E":".",
    "F":". . _ .",
    "G":"_ _ .",
    "H":". . . .",
    "I":". .",
    "J":". _ _ _",
    "K":"_ . _",
    "L":". _ . .",
    "M":"_ _",
    "N":"_ .",
    "O":"_ _ _",
    "P":". _ _ .",
    "Q":"_ _ . _",
    "R":". _ .",
    "S":". . .",
    "T":"_",
    "U":". . _",
    "V":". . . _",
    "W":". _ _",
    "X":"_ . . _",
    "Y":"_ . _ _",
    "Z":"_ _ . .",
}
dataBase1={
    ". _":"A",
    "_ . . .":"B",
    "_ . _ .":"C",
    "_ . .":"D",
    ".":"E",
    ". . _ .":"F",
    "_ _ .":"G",
    ". . . .":"H",
    ". .":"I",
    ". _ _ _":"J",
    "_ . _":"K",
    ". _ . .":"L",
    "_ _":"M",
    "_ .":"N",
    "_ _ _":"O",
    ". _ _ .":"P",
    "_ _ . _":"Q",
    ". _ .":"R",
    ". . .":"S",
    "_":"T",
    ". . _":"U",
    ". . . _":"V",
    ". _ _":"W",
    "_ . . _":"X",
    "_ . _ _":"Y",
    "_ _ . .":"Z",
}
if user_choice==1:
        print(morse_to_normal(user_msg,dataBase1,space_between_words,space_between_letters))
elif user_choice==2:
        print(normal_to_morse(user_msg.upper(),dataBase2))
